packages/terra-site/src/ThemeProvider.jsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+  /**
+   * The component(s) that will be wrapped by <ThemeProvider />
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Name of class for specified theme. e.g `terra-OSC-theme`
+   */
+  themeName: PropTypes.string.isRequired,
+};
+
+const ThemeProvider = ({ themeName, children, ...customProps }) => {
+  return <div className={themeName}>{children}</div>;
```

Comment:
>  I would rather us have a prop on the Terra Base that takes in a theme, and places that on the class

I'm fine having this logic in terra base.. but I do think solutions will need the ability for nested sub-themes.

> Or document the theme module in a way that tells the consumer just to attach the theme as a class to a component they are using.
The goal is to not have consumers aware of class names as they are an implementation detail.